streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java

```diff
@@ -239,10 +241,30 @@ public void shouldAddAndRemoveThreadsMultipleTimes() throws InterruptedException
             final CountDownLatch latch = new CountDownLatch(2);
             final Thread one = adjustCountHelperThread(kafkaStreams, 4, latch);
             final Thread two = adjustCountHelperThread(kafkaStreams, 6, latch);
-            two.start();
-            one.start();
-            latch.await(30, TimeUnit.SECONDS);
-            assertThat(kafkaStreams.metadataForLocalThreads().size(), equalTo(oldThreadCount));
+            Set<ThreadMetadata> threadMetadata = null;
+
+            AssertionError testError = null;
+            try {
+                two.start();
+                one.start();
+
+                assertTrue(latch.await(30, TimeUnit.SECONDS));
+                one.join();
```

Comment:
Doesn't this change the test, since you are joining `one` and `two` before asserting? Not strictly against that, just pointing it out because the PR description only mentions collecting extra logs and fixing a thread leak.

Reply:
> and fixing thread resource leak.

Don't think the `join()` order matters? But before, we did not join the threads at all?

Each thread count down the latch right before it exits, and we wait for the latch above.